### PR TITLE
[FIX] web_editor: prevent selection on sanitize

### PR DESCRIPTION
**Pull request by Goaman**
Before this commit, it was possible to set the selection of the document
whenever sanitizing an element that was not present in the dom.

task-2877501

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web_editor/static/lib/odoo-editor/src/utils/sanitize.js [hunk 1]
```diff
@@ -174,7 +179,7 @@ class Sanitize {
     }
 }
 
-export function sanitize(root) {
-    new Sanitize(root);
+export function sanitize(root, options) {
+    new Sanitize(root, options);
```
Comment (dmo-odoo) [inline addons/web_editor/static/lib/odoo-editor/src/utils/sanitize.js:183]: I fail to see when this is used.

**Comment by Goaman** [comment]
Up

File: addons/web_editor/static/lib/odoo-editor/src/utils/sanitize.js [hunk 2]
```diff
@@ -94,13 +94,17 @@ class Sanitize {
             return;
         }
 
+        const shouldSelect = node.isConnected;
+
         // Merge identical elements together
         while (areSimilarElements(node, node.previousSibling) && !isUnbreakable(node)) {
             getDeepRange(this.root, { select: true });
-            const restoreCursor = preserveCursor(this.root.ownerDocument);
+            const restoreCursor = shouldSelect && preserveCursor(this.root.ownerDocument);
```
[the remark this answers is not in the document]
Comment (Goaman) [reply addons/web_editor/static/lib/odoo-editor/src/utils/sanitize.js]: done

**Comment by dmo-odoo** [comment]
@Goaman Runbot is red for this branch.

**Comment by dmo-odoo** [comment]
@robodoo r+

**Pull request closed without merging**